indico/modules/events/abstracts/controllers/display.py

```diff
@@ -68,3 +70,37 @@ def _process(self):
                   .format(abstract.title, abstract.friendly_id), 'success')
             return jsonify_data(flash=False, redirect=url_for('.call_for_abstracts', self.event))
         return jsonify_template('events/abstracts/display/submission.html', event=self.event, form=form)
+
+
+class RHSubmitInvitedAbstract(RHAbstractBase):
+    def _check_access(self):
+        RHAbstractBase._check_access(self)
+        if self.abstract.state != AbstractState.invited:
+            raise Forbidden
+
+    def _check_abstract_protection(self):
+        if 'token' in request.args:
```

Comment:
see my suggestion from earlier: what about always using the token and not putting the normal ID in the URL at all for invited submissions?